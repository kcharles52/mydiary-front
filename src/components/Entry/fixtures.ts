export const props = {
  entry_id: 1,
  title: "Title 1",
  diaryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete:jest.fn()

};
