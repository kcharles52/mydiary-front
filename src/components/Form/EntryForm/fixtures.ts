export const EntryFormProps = {
  Edit: false,
  title: "string",
  diaryTitle: "string",
  date: "string",
  diaryBody: "string",
  onSubmit: jest.fn(),
  onCancel: jest.fn(),
  onChange: jest.fn()
};

  