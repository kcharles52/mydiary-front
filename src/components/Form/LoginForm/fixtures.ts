export const LoginFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  