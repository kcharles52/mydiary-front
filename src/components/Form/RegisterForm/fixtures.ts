export const RegisterFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  