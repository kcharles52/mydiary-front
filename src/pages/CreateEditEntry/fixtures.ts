export const props = {
  Edit: false,
  title: "Form title",
  diaryTitle: "Test Diary Title",
  date: "20-2-19",
  diaryBody: "Test Diary body",
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  addEntryAction: jest.fn(),
  editEntryAction: jest.fn(),
  getEntryAction: jest.fn(() =>
    Promise.resolve({
      entry: {
        entry_id: 1,
        diaryTitle: "Title 1",
        diaryEntryBody:
          "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
        date: "12/12/2018",
        delete: jest.fn()
      }
    })
  ),
  match: { params: { entry_id: "1" } },
  editEntry: {
    entry: {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  },
  newEntry: {},
  history: { push: () => {} },
  location: { pathname: "/addEntry" }
};

export const entry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
