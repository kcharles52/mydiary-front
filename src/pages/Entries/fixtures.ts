export const props = {
  getEntriesAction: jest.fn(),
  deleteEntryAction: jest.fn(),
  entries: [
    {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      diaryTitle: "Title 2",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  deleted: [],
  handlesDelete: jest.fn(),
  history: {push:jest.fn()},
  location: { pathname: "/home" },
  match: { url: "/home" }
};

export const fetchedEntries =[
  {
    entry_id: 1,
    diaryTitle: "Title 1",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  },
  {
    entry_id: 2,
    diaryTitle: "Title 2",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  }
]