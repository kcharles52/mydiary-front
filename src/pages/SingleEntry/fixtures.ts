export const props = {
  getEntryAction: (id: string) => Promise.resolve(),
  deleteEntryAction: (id: string) => Promise.resolve(),
  fetchedEntry: { entry: {} },
  match: { params: { entry_id: "1" } },
  history: { push: jest.fn() },
  location: { pathname: "/Entry/1" }
};

export const fetchedEntry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
