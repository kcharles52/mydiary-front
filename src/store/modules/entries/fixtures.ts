export const intialState = {
  entries: [],
  entry: {},
  new: {},
  Message: "",
  deleted: []
};

export const fetchedEntries = {
  entries: [
    {
      entry_id: 1,
      title: "Title 1",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      title: "Title 2",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  Message:"Successfully fetched entries"
};

export const editedEntry = {
    entry_id: 1,
      title: "Edited",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
}