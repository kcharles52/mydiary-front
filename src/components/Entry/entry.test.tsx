// react libraries
import * as React from "react";

// third party libraries
import { mount, shallow } from "enzyme";

//component
import { Entry } from ".";

//fixtures
import { props } from "./fixtures";

describe("Entry component", () => {
  let wrapper, shallowWrapper;
  beforeEach(() => {
    wrapper = mount(<Entry {...props} />);
    shallowWrapper = shallow(<Entry {...props} />);
  });

  afterEach(() => {
    wrapper = shallowWrapper = null;
  });

  it("should render entry propery", () => {
    expect(shallowWrapper).toMatchSnapshot();
    expect(shallowWrapper.find("button").length).toBe(2);
  });

  it("should simulate delete", () => {
    wrapper
      .find("button")
      .last()
      .simulate("click");
      expect(props.delete).toBeCalled();
  });
});
