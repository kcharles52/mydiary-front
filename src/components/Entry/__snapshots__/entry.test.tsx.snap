// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component should render entry propery 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Entry
    date="12/12/2018"
    delete={[MockFunction]}
    diaryBody="Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente "
    entry_id={1}
    title="Title 1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="DiaryCard"
      >
        <div
          className="DiaryCardHeader"
        >
          <a
            href="/Entry/1"
          >
            <h2 />
          </a>
          <h3>
            Diary date: 
            12/12/2018
             
          </h3>
        </div>
        <div
          className="DiaryCardBody"
        />
        <div
          className="DiaryCardButtons"
        >
          <button
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="DiaryCardHeader"
          >
            <a
              href="/Entry/1"
            >
              <h2 />
            </a>
            <h3>
              Diary date: 
              12/12/2018
               
            </h3>
          </div>,
          <div
            className="DiaryCardBody"
          />,
          <div
            className="DiaryCardButtons"
          >
            <button
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              onClick={[Function]}
            >
              Delete
            </button>
          </div>,
        ],
        "className": "DiaryCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="/Entry/1"
              >
                <h2 />
              </a>,
              <h3>
                Diary date: 
                12/12/2018
                 
              </h3>,
            ],
            "className": "DiaryCardHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2 />,
                "href": "/Entry/1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Diary date: ",
                  "12/12/2018",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Diary date: ",
                "12/12/2018",
                " ",
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "DiaryCardBody",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                Edit
              </button>,
              <button
                onClick={[Function]}
              >
                Delete
              </button>,
            ],
            "className": "DiaryCardButtons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="DiaryCard"
        >
          <div
            className="DiaryCardHeader"
          >
            <a
              href="/Entry/1"
            >
              <h2 />
            </a>
            <h3>
              Diary date: 
              12/12/2018
               
            </h3>
          </div>
          <div
            className="DiaryCardBody"
          />
          <div
            className="DiaryCardButtons"
          >
            <button
              onClick={[Function]}
            >
              Edit
            </button>
            <button
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="DiaryCardHeader"
            >
              <a
                href="/Entry/1"
              >
                <h2 />
              </a>
              <h3>
                Diary date: 
                12/12/2018
                 
              </h3>
            </div>,
            <div
              className="DiaryCardBody"
            />,
            <div
              className="DiaryCardButtons"
            >
              <button
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                onClick={[Function]}
              >
                Delete
              </button>
            </div>,
          ],
          "className": "DiaryCard",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="/Entry/1"
                >
                  <h2 />
                </a>,
                <h3>
                  Diary date: 
                  12/12/2018
                   
                </h3>,
              ],
              "className": "DiaryCardHeader",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h2 />,
                  "href": "/Entry/1",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h2",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Diary date: ",
                    "12/12/2018",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Diary date: ",
                  "12/12/2018",
                  " ",
                ],
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "DiaryCardBody",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  onClick={[Function]}
                >
                  Edit
                </button>,
                <button
                  onClick={[Function]}
                >
                  Delete
                </button>,
              ],
              "className": "DiaryCardButtons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Edit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
